client/state/data-layer/wpcom/sites/admin-menu/index.js

```diff
@@ -17,8 +18,43 @@ export const requestFetchAdminMenu = ( action ) =>
 		action
 	);
 
-export const handleSuccess = ( { siteId }, menuData ) => {
-	return receiveAdminMenu( siteId, menuData );
+const sanitizeUrl = ( url, site ) => {
+	const isSafeInternalUrl = new RegExp( '^/' ).test( url );
+	const isSafeSiteDomainUrl = new RegExp( `^https?://${ site?.domain }` ).test( url );
+	if ( isSafeInternalUrl || isSafeSiteDomainUrl ) {
+		return url;
+	}
+
+	return '';
+};
+
+const sanitizeMenuItem = ( menuItem, site ) => {
```

Comment:
Shall we write some unit tests for this, or creating a follow up issue to do so?